compiler/src/dotty/tools/dotc/semanticdb/ExtractSemanticDB.scala

```diff
@@ -502,7 +504,7 @@ class ExtractSemanticDB extends Phase:
         }).toMap
     end findGetters
 
-    private def adjustSpanToName(span: Span, qualSpan: Span, name: Name)(using Context) =
+    private def adjustSpanToName(span: Span, qualSpan: Span, name: Name, source: SourceFile) =
```

Comment:
Seems like we should remove the `inline private def source(using Context) = ctx.compilationUnit.source` from this file to prevent accidentally using the wrong source.

Reply:
I've updated the code so that the Tree is passed implicitly wherever we require a source

Reply:
Hmm this might work but it's kind of an unusual pattern and still seems easy to do the wrong thing by forgetting to pass `given` somewhere. My suggestion would be to pass it explicitly.

Reply:
ok I've updated to use explicit passing